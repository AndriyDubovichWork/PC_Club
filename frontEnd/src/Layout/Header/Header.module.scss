.Header {
  height: 10%;

  width: 80vw;
  margin: 0 auto;

  display: flex;
  flex-direction: row;
  justify-content: space-between;

  .Logo {
    display: flex;
    flex-direction: row;
    cursor: pointer;

    text-decoration: none;
    color: #000;

    .Img {
      height: 10vh;
      width: 10vh;
    }
    .Name {
      display: flex;
      flex-direction: column;
    }
  }
  .LoginRegister {
    display: flex;
    align-items: center;
  }
}
