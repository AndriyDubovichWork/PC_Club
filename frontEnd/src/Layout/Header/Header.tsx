import React, { useContext, useState } from 'react';
import style from './Header.module.scss';
import NavBar from './NavBar/NavBar';
import ButtonMUI from '@mui/material/Button';
import { RefContext } from '../../Context/RefContext';
import { NavLink } from 'react-router-dom';
import useIsPCScreen from '../../Hooks/useIsPCScreen';
import MenuIcon from '@mui/icons-material/Menu';
import Button from '../../Styled/Button/Button';
import { IsAuthorisedContext } from '../../Context/IsAuthorisedContext';
const Header = () => {
  const { HeaderRef } = useContext(RefContext);
  const { isPC } = useIsPCScreen();

  const [isNavBarOpen, setIsNavBarOpen] = useState(false);
  const { isAuthorised, setIsAuthorised } = useContext(IsAuthorisedContext);

  return (
    <header className={style.Header} ref={HeaderRef}>
      <NavLink to='/PC_Club' className={style.Logo}>
        <img
          loading='lazy'
          src={require('./assets/PCIcon.png')}
          alt='logo'
          className={style.Img}
        />
        <div className={style.Name}>
          <h1>PC</h1>
          <h2>CLUB</h2>
        </div>
      </NavLink>
      <NavBar isNavBarOpen={isNavBarOpen} setIsNavBarOpen={setIsNavBarOpen} />

      {isPC ? (
        <>
          <div className={style.LoginRegister}>
            {!isAuthorised ? (
              <>
                <NavLink to='/PC_Club/login'>
                  <ButtonMUI
                    variant='text'
                    sx={{
                      marginRight: '2vw',
                      color: '#000',
                      '&:hover': {
                        color: '#ff4d30',
                      },
                    }}
                  >
                    Log In
                  </ButtonMUI>
                </NavLink>
                <NavLink to='/PC_Club/register'>
                  <Button>Register</Button>
                </NavLink>
              </>
            ) : (
              <ButtonMUI
                variant='text'
                sx={{
                  marginRight: '2vw',
                  color: '#000',
                  '&:hover': {
                    color: '#ff4d30',
                  },
                }}
                onClick={() => setIsAuthorised(false)}
              >
                Log Out
              </ButtonMUI>
            )}
          </div>
        </>
      ) : (
        <ButtonMUI onClick={() => setIsNavBarOpen(true)}>
          <MenuIcon sx={{ color: '#ff4d30' }} />
        </ButtonMUI>
      )}
    </header>
  );
};

export default Header;
