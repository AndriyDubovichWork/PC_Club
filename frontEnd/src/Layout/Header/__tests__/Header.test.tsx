import { render, screen, cleanup } from '@testing-library/react';
import Header from '../Header';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Header match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club']}>
        <Routes>
          <Route path='/PC_Club' element={<Header />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Header contains logo', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club']}>
      <Routes>
        <Route path='/PC_Club' element={<Header />} />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('CLUB')).toBeInTheDocument();
});
