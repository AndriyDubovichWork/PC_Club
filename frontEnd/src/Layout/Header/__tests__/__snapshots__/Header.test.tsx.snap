// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header match snapshot 1`] = `
<header
  className="Header"
>
  <a
    aria-current="page"
    className="Logo active"
    href="/PC_Club"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="Img"
      loading="lazy"
      src="PCIcon.png"
    />
    <div
      className="Name"
    >
      <h1>
        PC
      </h1>
      <h2>
        CLUB
      </h2>
    </div>
  </a>
  <nav
    className="NavBar"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-pt19bo-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</header>
`;
